controllers/clusterca_to_secret.go

```diff
@@ -0,0 +1,190 @@
+package controllers
+
+import (
+	"context"
+	"encoding/hex"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	v12 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+const (
+	Namespace  = "global-trust-certificates"
+	SecretName = "cluster-ca-secret"
+)
+
+func ClusterCaToSecret(client client.Client) {
+	log := log.Log.WithName("ClusterCAToSecret")
+
+	ticker := time.NewTicker(20 * time.Second)
+	quit := make(chan struct{})
+	for {
+		select {
+		case <-ticker.C:
+
+			isPresent, err := checkIfSecretPresent(context.Background(), client, log)
+
+			if isPresent {
+				// check the hash of the secret data with cert.
+				sec, err := getSecret(context.Background(), client, log)
```

Comment:
For this error case and the others below it, should the error be handled beyond logging it. Mainly, should execution continue with the statements that follow it. Inside a select-case, you can use break to end the case and wait for the next tick case.

Reply:
After reviewing the further cases of error logging, we should separate the entire ticker case into a single function that returns an error. This is so we can wrap errors so its easier to trace where they are similar to a stack trace. Most of the functions in the CertificatePackage controller show example of this. Using this pattern, there would be a single error line that says something like `failed to do X: failed to do Y...`. In that example something called function X and function X tried to do Y and failed. Kind of like bread crumbs. Its a common pattern used by many go projects. The syntax when you have err != nil, is to return fmt.Errorf("failed something: %s", err.Error())